### PR TITLE
Add v2 migrations steps

### PR DESCRIPTION
**Pull request by pmalek-sumo**
###### Description

Add v2 migrations steps

###### Testing performed

- [ ] ci/build.sh
- [ ] Redeploy fluentd and fluentd-events pods
- [ ] Confirm events, logs, and metrics are coming in

### CODE REVIEW

File: deploy/docs/v2_migration_doc.md [hunk 1]
```diff
@@ -76,3 +76,65 @@ the exact steps for migration.
 ## How to upgrade
 
 **Note: The below steps are using Helm 3. Helm 2 is not supported.**
+
+### 1. Upgrade to helm chart version `v1.13.2`
```
Comment (sumo-drosiek) [inline deploy/docs/v2_migration_doc.md:80]: `v1.3.2` I suppose

Comment (pmalek-sumo) [reply deploy/docs/v2_migration_doc.md]: yes ofc..... fixed

File: deploy/docs/v2_migration_doc.md [hunk 2]
```diff
@@ -76,3 +76,65 @@ the exact steps for migration.
...
+  with the below command:
+
+  ```bash
+  helm get values <RELEASE-NAME> > current_values.yaml
+  ```
+
+- Run `curl` the upgrade script as follows:
+
+  ```bash
+  curl -LJO https://raw.githubusercontent.com/SumoLogic/sumologic-kubernetes-collection/main/deploy/helm/sumologic/upgrade-2.0.0.sh
```
Comment (sumo-drosiek) [inline deploy/docs/v2_migration_doc.md:127]: please remember to upgrade it when the release 2.0 branch will be created

Comment (pmalek-sumo) [reply deploy/docs/v2_migration_doc.md:127]: https://github.com/SumoLogic/sumologic-kubernetes-collection/issues/1174

File: deploy/docs/v2_migration_doc.md [hunk 3]
```diff
@@ -82,3 +82,65 @@ the exact steps for migration.
...
+
+```bash
+helm repo update
+```
+
+For the users who are not already on `v1.3.2` of the helm chart, please upgrade
+to that version first by running the below command.
+
+```bash
+helm upgrade collection sumologic/sumologic --reuse-values --version=1.3.2
```
Comment (sumo-drosiek) [inline deploy/docs/v2_migration_doc.md:98]: could we test against `--reuse-values`? AFAIK changes in values.yaml won't be applied

**Review by sumo-drosiek — APPROVED** [review]

File: deploy/docs/v2_migration_doc.md [hunk 4]
```diff
@@ -82,3 +82,65 @@ the exact steps for migration.
...
+- Run `curl` the upgrade script as follows:
+
+  ```bash
+  curl -LJO https://raw.githubusercontent.com/SumoLogic/sumologic-kubernetes-collection/main/deploy/helm/sumologic/upgrade-2.0.0.sh
+  ```
+
+- Run the upgrade script on the above file with the below command.
+
+  ```bash
+  ./upgrade-2.0.0.sh current_values.yaml
```
Comment (kkujawa-sumo) [inline deploy/docs/v2_migration_doc.md:139]: Maybe before this step we should add `chmod +x upgrade-2.0.0.sh`. It is quite obvious but usually this step appears in instructions, e.g. [here](https://kubernetes.io/docs/tasks/tools/install-kubectl/#install-kubectl-binary-with-curl-on-linux) and [here](https://helm.sh/docs/intro/install/#from-script)

Comment (kkujawa-sumo) [reply deploy/docs/v2_migration_doc.md:139]: I'm testing on vagrant VM created from `release-v1.3` branch, commit: 2fac768fb.
I see following error:
```
./upgrade-2.0.0.sh current_values.yaml
[INFO]    Mapping fluentd.image.repository into:
[INFO]    Mapping fluentd.image.tag into:
[INFO]    Mapping image.repository into:
[INFO]    - fluentd.image.repository
[INFO]    - sumologic.setup.job.image.repository
[INFO]    Mapping image.tag into:
[INFO]    - fluentd.image.tag
[INFO]    - sumologic.setup.job.image.tag

[INFO]    Migrating prometheus-config-reloader container to config-reloader in prometheusSpec
Error: unknown flag: --arrays
```

yq version: 
```
$ yq --version
yq version 3.3.0
```

Comment (pmalek-sumo) [reply deploy/docs/v2_migration_doc.md:139]: Good points

Comment (pmalek-sumo) [reply deploy/docs/v2_migration_doc.md:139]: You'll need `yq` at least at version `3.4.0` 

The following handles that at VM provision time but you can run it manually

https://github.com/SumoLogic/sumologic-kubernetes-collection/blob/63c6cf3415ed5be396834d527f51612fa8382f66/vagrant/provision.sh#L53-L56

File: deploy/docs/v2_migration_doc.md [hunk 5]
```diff
@@ -82,3 +82,65 @@ the exact steps for migration.
 ## How to upgrade
 
 **Note: The below steps are using Helm 3. Helm 2 is not supported.**
+
+### 1. Upgrade to helm chart version `v1.3.2`
+
+Run the below command to fetch the latest helm chart:
+
+```bash
+helm repo update
```
Comment (kkujawa-sumo) [inline deploy/docs/v2_migration_doc.md:91]: I think that before this step we can add:
Please make sure that you have sumologic helm repo configured, check this using `helm repo list`. If sumologic helm repo is not configured use `helm repo add sumologic https://sumologic.github.io/sumologic-kubernetes-collection`

Maybe it'll make someone's life easier ;) When I deployed collection 1.3.2 from sources I don't have any helm repo.

File: deploy/docs/v2_migration_doc.md (hunk 4 above)
Comment (kkujawa-sumo) [reply deploy/docs/v2_migration_doc.md]: I'm testing from `release-v1.3` branch,  https://github.com/SumoLogic/sumologic-kubernetes-collection/blob/release-v1.3/vagrant/provision.sh#L56

May we mention required `yq` version in this instruction?

Comment (pmalek-sumo) [reply deploy/docs/v2_migration_doc.md]: Ok, apparently I missed and incorrectly changed the v2 update script

it should be fixed (i.e. user will get a notification about required `yq` version) via:

https://github.com/SumoLogic/sumologic-kubernetes-collection/pull/1172/files#diff-e010a115e1f3b27c1977c58d40e6aad5860cb28d7beb501161fb7d4a9358e6cc

File: deploy/docs/v2_migration_doc.md (hunk 5 above)
Comment (pmalek-sumo) [reply deploy/docs/v2_migration_doc.md:113]: Added

File: deploy/docs/v2_migration_doc.md [hunk 6]
```diff
@@ -82,3 +82,87 @@ the exact steps for migration.
...
+- Run the upgrade script on the above file with the below command.
+
+  ```bash
+  chmod +x upgrade-2.0.0.sh && ./upgrade-2.0.0.sh current_values.yaml
+  ```
+
+- At this point, users can then run:
+
+  ```bash
+  helm upgrade collection sumologic/sumologic --version=2.0.0 -f new_values.yaml
```
Comment (kkujawa-sumo) [inline deploy/docs/v2_migration_doc.md:167]: As 2.0.0 release is not available yet. I tested with dev version - 2.0.0-dev.0-133-g6e362ae, so I added dev helm repo:
```helm repo list
NAME         	URL
sumologic    	https://sumologic.github.io/sumologic-kubernetes-collection
sumologic-dev	https://sumologic.github.io/sumologic-kubernetes-collection/dev
```
and I see following error:
```
$ helm upgrade collection sumologic-dev/sumologic --version=2.0.0-dev.0-133-g6e362ae -f new_values.yaml -n sumologic
Error: UPGRADE FAILED: cannot patch "collection-fluent-bit" with kind DaemonSet: DaemonSet.apps "collection-fluent-bit" is invalid: spec.selector: Invalid value: v1.LabelSelector{MatchLabels:map[string]string{"app.kubernetes.io/instance":"collection", "app.kubernetes.io/name":"fluent-bit"}, MatchExpressions:[]v1.LabelSelectorRequirement(nil)}: field is immutable && cannot patch "collection-sumologic-fluentd-events" with kind StatefulSet: StatefulSet.apps "collection-sumologic-fluentd-events" is invalid: spec: Forbidden: updates to statefulset spec for fields other than 'replicas', 'template', and 'updateStrategy' are forbidden && cannot patch "collection-sumologic-fluentd-metrics" with kind StatefulSet: StatefulSet.apps "collection-sumologic-fluentd-metrics" is invalid: spec: Forbidden: updates to statefulset spec for fields other than 'replicas', 'template', and 'updateStrategy' are forbidden && cannot patch "collection-sumologic-fluentd-logs" with kind StatefulSet: StatefulSet.apps "collection-sumologic-fluentd-logs" is invalid: spec: Forbidden: updates to statefulset spec for fields other than 'replicas', 'template', and 'updateStrategy' are forbidden
```

Comment (pmalek-sumo) [reply deploy/docs/v2_migration_doc.md:167]: So this is most likely connected to labels and/or statefulset's volume claim when `fluentd.persistence` is enabled.

https://github.com/SumoLogic/sumologic-kubernetes-collection/blob/d1ed616c0cb4ae3c46c258eacf48c8eaaac6b22b/deploy%2Fhelm%2Fsumologic%2Ftemplates%2Fstatefulset.yaml#L151-L161

I'll add warning to upgrade 2.0.0 script, make it leave persistence as disabled if user set it as such and add short doc about how to enable it.

**Pull request merged**